editor/components/writing-flow/index.js

```diff
@@ -261,6 +261,14 @@ class WritingFlow extends Component {
 			const blockContainer = this.container.querySelector( `[data-block="${ this.props.selectedBlockUID }"]` );
 			if ( blockContainer && ! blockContainer.contains( document.activeElement ) ) {
 				const target = this.getInnerTabbable( blockContainer, this.props.initialPosition === -1 );
+
+				// Avoid selecting the target if it's:
+				//  - The default block appender (or generally uneditable)
+				//  - Within a nested block
+				if ( target.readOnly || target.closest( '[data-block]' ) !== blockContainer ) {
```

Comment:
This breaks some behaviors:

 - splitting inside container blocks like "columns".
 - selecting an image and hitting Enter inside container blocks

Reply:
Hmm, I can confirm this, though I wouldn't have expected it, since these behaviors should be finding inputs within the selected block, at which point it should be finding the selected nested's inputs correctly. Will investigate...